src/libutil/hash.cc

```diff
@@ -50,21 +50,14 @@ bool Hash::operator == (const Hash & h2) const
 }
 
 
-bool Hash::operator != (const Hash & h2) const
+std::strong_ordering Hash::operator <=> (const Hash & h) const
 {
-    return !(*this == h2);
-}
-
-
-bool Hash::operator < (const Hash & h) const
-{
-    if (hashSize < h.hashSize) return true;
-    if (hashSize > h.hashSize) return false;
+    if (auto cmp = algo <=> h.algo; cmp != 0) return cmp;
```

Comment:
Including the algo is good, but checking it first changes the ordering.
I don't know if anything relies on the ordering for stability, in the sense of reproducibility, but if it does, this could lead to a bug (potentially subtle because a lot is sha256, but not everything).
Unless we want to figure out where hashes are ordered, it's best to move the algo comparison to the end.